packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<{ accounts: string[], provider: Web3Provider | null }>;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+};
+
+const PassportContext = createContext<PassportContextType>({
+  passportInstance: {} as Passport,
+  isLoggedIn: false,
+  login: () => { throw new Error('login must be used within a PassportProvider'); },
+  logout: () => { throw new Error('logout must be used within a PassportProvider'); },
+  loginWithoutWallet: () => { throw new Error('loginWithoutWallet must be used within a PassportProvider'); },
+  loginWithEthersjs: () => { throw new Error('loginWithEthersjs must be used within a PassportProvider'); },
+  isLoading: false,
+  error: null,
+  accounts: [],
+});
+
+type PassportProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function PassportProvider({ children, config }: PassportProviderProps) {
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [accounts, setAccounts] = useState<string[]>([]);
+  const [error, setError] = useState<Error | null>(null);
```

Comment:
Hmmm no... I don't think you want to say this is Error or null when you're putting `any` in to it.

Reply:
casted to Error in catch statements